lib/ts/recipe/session/components/themes/accessDeniedScreenTheme/index.tsx

```diff
@@ -17,21 +18,21 @@ import type { FC } from "react";
 const AccessDeniedScreen: FC<AccessDeniedThemeProps> = (props) => {
     const userContext = useUserContext();
     const t = useTranslation();
+
     const onLogout = async () => {
-        try {
-            await props.recipe.signOut({ userContext });
-        } catch (error) {
-        } finally {
-            await SuperTokens.getInstanceOrThrow().redirectToAuth({
-                show: "signin",
-                redirectBack: false,
-            });
-        }
+        await props.recipe.signOut({ userContext });
+        await SuperTokens.getInstanceOrThrow().redirectToAuth({
+            show: "signin",
+            redirectBack: false,
+        });
     };
 
     const onBackButtonClicked = () => {
-        if (props.history.back !== undefined) {
-            return props.history.back();
+        if (props.history === undefined) {
```

Comment:
- Please add an explanatory comment to say which branch handles which case.
- Add an end-to-end test for each branch please